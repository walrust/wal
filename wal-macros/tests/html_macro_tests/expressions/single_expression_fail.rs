use wal_macros::html;

struct TestNonDisplayStruct {
    field: i32,
}

fn main() {
    html! { () };
    html! { invalid_reference };
    html! { TestNonDisplayStruct { field: 15 } };
    let node = || TestNonDisplayStruct { field: 15 };
    html! { node() };
}
