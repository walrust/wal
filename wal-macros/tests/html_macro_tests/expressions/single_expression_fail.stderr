error[E0425]: cannot find value `invalid_reference` in this scope
 --> tests/html_macro_tests/expressions/single_expression_fail.rs:7:13
  |
7 |     html! { invalid_reference };
  |             ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0277]: `()` doesn't implement `std::fmt::Display`
 --> tests/html_macro_tests/expressions/single_expression_fail.rs:6:13
  |
6 |     html! { () };
  |             ^^ `()` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `()`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = help: the following other types implement trait `From<T>`:
            <VNode as From<T>>
            <VNode as From<VComponent>>
            <VNode as From<VElement>>
            <VNode as From<VList>>
            <VNode as From<VText>>
  = note: required for `()` to implement `ToString`
  = note: required for `VNode` to implement `From<()>`

error[E0277]: `NonDisplayStruct` doesn't implement `std::fmt::Display`
 --> tests/html_macro_tests/expressions/single_expression_fail.rs:8:13
  |
8 |     html! { NonDisplayStruct };
  |             ^^^^^^^^^^^^^^^^ `NonDisplayStruct` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `NonDisplayStruct`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = help: the following other types implement trait `From<T>`:
            <VNode as From<T>>
            <VNode as From<VComponent>>
            <VNode as From<VElement>>
            <VNode as From<VList>>
            <VNode as From<VText>>
  = note: required for `NonDisplayStruct` to implement `ToString`
  = note: required for `VNode` to implement `From<NonDisplayStruct>`

error[E0277]: `NonDisplayStruct` doesn't implement `std::fmt::Display`
  --> tests/html_macro_tests/expressions/single_expression_fail.rs:10:13
   |
10 |     html! { node() };
   |             ----^^
   |             |
   |             `NonDisplayStruct` cannot be formatted with the default formatter
   |             required by a bound introduced by this call
   |
   = help: the trait `std::fmt::Display` is not implemented for `NonDisplayStruct`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = help: the following other types implement trait `From<T>`:
             <VNode as From<T>>
             <VNode as From<VComponent>>
             <VNode as From<VElement>>
             <VNode as From<VList>>
             <VNode as From<VText>>
   = note: required for `NonDisplayStruct` to implement `ToString`
   = note: required for `VNode` to implement `From<NonDisplayStruct>`
